### ISSUE TITLE
Suggestion: In regards to Windows UI & GUI choices

### ISSUE THREAD

**Issue by Not-Alaksa-Airlines** [face-hh/webx#101]
### Prologue

![image](https://github.com/face-hh/webx/assets/171449411/b68b6d27-39a1-46e2-8e88-18533b2f3286)
_Fig. 1_

As you can see (in Figure 1), this interface is plain and empty. My suggestion includes adding tabs to allow multiple sites to be open at once, as well as new choices to make the GUI more Windows-friendly on the Windows release.  Firstly, let's tackle the Windows-friendly GUI.

### Windows-friendly GUI

![image](https://github.com/face-hh/webx/assets/171449411/01b52e04-2a33-4092-b748-b04dcb713ed8)
_Fig. 2_

![image](https://github.com/face-hh/webx/assets/171449411/7831df71-46bd-416a-a0e3-e1d0e4da5c99)
_Fig. 3_

As you can see here (in Figure 2), the GUI most applications in Windows use is very plain and simple, though effective due to their full extension to the bottom of the header. It would make the most sense to replace the current title bar buttons of WebX Napture (as seen in Figure 3) with ones much more like the common Windows title bar, to blend more with other applications. However, if the other buttons are preferred or are more normal in the Linux release, this change could be available only for the Windows release.

### Tabs - Various possibilities

![image](https://github.com/face-hh/webx/assets/171449411/7a511f0e-1a7e-4808-9af8-1b661ac5c9c0)
_Fig. 4_

![image](https://github.com/face-hh/webx/assets/171449411/9f5fdd4d-d638-4b58-abd7-07a888e9e30e)
_Fig. 5_

As displayed (in Figure 4), the central header is very empty, which can bring many possibilities for tabs. The ability to open multiple websites at once could pave the way for many other features and a more general sense of normalcy. The first suggestion of mine, which I prefer, is that we use the space beyond the address bar as a place to put tabs, these tabs would behave as normal. However, special features are always up to interpretation. My second suggestion is a lot more common. This suggestion reflects the typical design of browsers like Google Chrome (as seen in Figure 5.) The search bar would be stretched across the whole screen, and tabs would be put above it.

### Conclusion
These ideas all come with pros and cons, but I believe they are well-rounded concepts. As always, I shall leave my ideas up to the community of other people who are also using this application.

**TL;DR**
Title bar buttons, more like those of other Windows applications, and tabs.

**Comment by GStudiosX2** [comment]
I think to change the header bar you need to install a custom gtk theme

**Comment by Not-Alaksa-Airlines** [comment]
It's supposed to be a universal feature, to make it better. The whole point is that you shouldn't have to change the theme to have continuity. It's of course subjective on what makes something better, but in my opinion, it will be better this way, for the Windows release at least.

**Comment by GStudiosX2** [comment]
> It's supposed to be a universal feature, to make it better. The whole point is that you shouldn't have to change the theme to have continuity. It's of course subjective on what makes something better, but in my opinion, it will be better this way, for the Windows release at least.

GTK isn't meant for windows

**Comment by Not-Alaksa-Airlines** [comment]
Has something like that stopped people before? Why does it have to now?

**Comment by Not-Alaksa-Airlines** [comment]
> Has something like that stopped people before? Why does it have to now?

Especially with something as trivial as potentially spending a week or two trying to change one little thing, comparatively trivial at least.

**Comment by GStudiosX2** [comment]
> Has something like that stopped people before? Why does it have to now?

I'm not saying it's impossible probably could preinstall some theme or something I haven't really used GTK that much I only messed with a bit I'm pretty sure it's not customizable without a theme though unless face makes his own HeaderBar

**Comment by GStudiosX2** [comment]
I do think it would be nice to be consistent with other apps on windows and I would also like tabs but i don't think too much about it not being the same button styles as other apps

**Comment by Not-Alaksa-Airlines** [comment]
> I do think it would be nice to be consistent with other apps on windows and I would also like tabs but i don't think too much about it not being the same button styles as other apps

I at least need the buttons to extend to the bottom of the header.

**Comment by RealPacket** [comment]
> ### Prologue
> ![image](https://private-user-images.githubusercontent.com/171449411/335828272-b68b6d27-39a1-46e2-8e88-18533b2f3286.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MTQ0MzEsIm5iZiI6MTcxODQxNDEzMSwicGF0aCI6Ii8xNzE0NDk0MTEvMzM1ODI4MjcyLWI2OGI2ZDI3LTM5YTEtNDZlMi04ZTg4LTE4NTMzYjJmMzI4Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwMTE1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YWQ4ZTdhOThkYzNjMDlkMjk5YThiNmU1ZTUwMGU5ZTQ4ODUxYjg3NDY4MDk5ZmJkNmM2NjU5ZGIzNTlmNDQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wA81eCuFnqyKfySMI5IQBwGqiFqG5TNmVFWFqP_MJS8) _Fig. 1_
> 
> As you can see (in Figure 1), this interface is plain and empty. My suggestion includes adding tabs to allow multiple sites to be open at once, as well as new choices to make the GUI more Windows-friendly on the Windows release. Firstly, let's tackle the Windows-friendly GUI.
> 
> ### Windows-friendly GUI
> ![image](https://private-user-images.githubusercontent.com/171449411/335828313-01b52e04-2a33-4092-b748-b04dcb713ed8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MTQ0MzEsIm5iZiI6MTcxODQxNDEzMSwicGF0aCI6Ii8xNzE0NDk0MTEvMzM1ODI4MzEzLTAxYjUyZTA0LTJhMzMtNDA5Mi1iNzQ4LWIwNGRjYjcxM2VkOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwMTE1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDVkZjRhNjNjY2Q4NDkwOTUxMDg3MzNhYTJlZGU2NjExYjVmZjMyNjhlODdkMWZmMzQxMTU5OTg1OGFkY2Q4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DY_JDC_ZRJm0UtNDzAG77zkNsuzakM-UZecgp2FJGBA) _Fig. 2_
> 
> ![image](https://private-user-images.githubusercontent.com/171449411/335828375-7831df71-46bd-416a-a0e3-e1d0e4da5c99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MTQ0MzEsIm5iZiI6MTcxODQxNDEzMSwicGF0aCI6Ii8xNzE0NDk0MTEvMzM1ODI4Mzc1LTc4MzFkZjcxLTQ2YmQtNDE2YS1hMGUzLWUxZDBlNGRhNWM5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwMTE1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYmU5YmI0YzJmNTc4MmU3MTY4MDYxM2E2N2JjNzM3NDc4MjI5MTc2NGFhYzM4NWRiZmFkMmZlMTNmMzM5MWZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TL_ak2ascDsIOJAZedL9yFkI_Pm5GuavqRdTHHG6tEQ) _Fig. 3_
> 
> As you can see here (in Figure 2), the GUI most applications in Windows use is very plain and simple, though effective due to their full extension to the bottom of the header. It would make the most sense to replace the current title bar buttons of WebX Napture (as seen in Figure 3) with ones much more like the common Windows title bar, to blend more with other applications. However, if the other buttons are preferred or are more normal in the Linux release, this change could be available only for the Windows release.
> 
> ### Tabs - Various possibilities
> ![image](https://private-user-images.githubusercontent.com/171449411/335828696-7a511f0e-1a7e-4808-9af8-1b661ac5c9c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MTQ0MzEsIm5iZiI6MTcxODQxNDEzMSwicGF0aCI6Ii8xNzE0NDk0MTEvMzM1ODI4Njk2LTdhNTExZjBlLTFhN2UtNDgwOC05YWY4LTFiNjYxYWM1YzljMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwMTE1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZGZhYzc4MDRiNjIxZTU0ZDU0MWRjNDJlZjVkODg5YTg3NTE0NzA1MGZmODY2M2NmZGM0NDJjZTNlZGM2ODQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HpKYmgyze8AO6dVYcdLESNrAk4vOjFll7NjhWaMBx5c) _Fig. 4_
> 
> ![image](https://private-user-images.githubusercontent.com/171449411/335828947-9f5fdd4d-d638-4b58-abd7-07a888e9e30e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MTQ0MzEsIm5iZiI6MTcxODQxNDEzMSwicGF0aCI6Ii8xNzE0NDk0MTEvMzM1ODI4OTQ3LTlmNWZkZDRkLWQ2MzgtNGI1OC1hYmQ3LTA3YTg4OGU5ZTMwZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwMTE1MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZjUxYzExNTE1NjZlNDMzZWQxOTBlNjUzMjA0NThiZWQ3MzU1MDc4N2UwYjdmZTMyYzkxMGQ0ZDA4NTEzMTUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3dCuKAkntVUxCZltF1--jL4_TAXJJGOwqby5MwUi-cg) _Fig. 5_
> 
> As displayed (in Figure 4), the central header is very empty, which can bring many possibilities for tabs. The ability to open multiple websites at once could pave the way for many other features and a more general sense of normalcy. The first suggestion of mine, which I prefer, is that we use the space beyond the address bar as a place to put tabs, these tabs would behave as normal. However, special features are always up to interpretation. My second suggestion is a lot more common. This suggestion reflects the typical design of browsers like Google Chrome (as seen in Figure 5.) The search bar would be stretched across the whole screen, and tabs would be put above it.
> 
> ### Conclusion
> These ideas all come with pros and cons, but I believe they are well-rounded concepts. As always, I shall leave my ideas up to the community of other people who are also using this application.
> 
> **TL;DR** Title bar buttons, more like those of other Windows applications, and tabs.

He's tried to add tabs support, but it never made it into releases because of a bug or something.
